### PR TITLE
Improve partial evaluation

### LINKED ISSUE #3224: Partially evaluation for xor expressions with consts
Currently this works properly:
`((x | 1) | 3) | 5` -> `x | 7`
`((x & 5) & 3) & 1` -> `x & 1`

But folding xor constants as:
`((x ^ -2) ^ -1) ^ -5` -> `x ^ -5`
It seems **doesn't work** for now!

@kripken could you point me which part of the code to look at, so that I could fix it?

### PR DESCRIPTION
**Pull request by MaxGraey**
Also handle 64-bit and, or & xor operations

Fix #3224

### CODE REVIEW

**Comment by MaxGraey** [comment]
fuzzed
```
Invocations so far:
   FuzzExec: 10237
   CompareVMs: 2806
   CheckDeterminism: 883
   Wasm2JS: 2595
   Asyncify: 2700

ITERATION: 12061
```

File: test/passes/optimize-instructions_all-features.wast [hunk 1]
```diff
@@ -2,6 +2,15 @@
   (memory 0)
   (type $0 (func (param i32 i64)))
   (func $f (type $0) (param $i1 i32) (param $i2 i64)
+    (drop
```
Comment (tlively) [inline test/passes/optimize-instructions_all-features.wast:5]: Can you add tests for the new & and | cases as well?

**Review by tlively — COMMENTED** [review]
LGTM with additional tests.

**Comment by MaxGraey** [comment]
Recent commit add the same for `Mul` operations. `DivS` and `DivU` will be added in future due to they have some edge cases with overflows and division by zero.

**Comment by MaxGraey** [comment]
Refuzzed:
```
Invocations so far:
   FuzzExec: 8674
   CompareVMs: 2333
   CheckDeterminism: 753
   Wasm2JS: 2051
   Asyncify: 2216

ITERATION: 10174
```

**Review by tlively — APPROVED** [review]

**Pull request merged**